### PR TITLE
GAP: switch some code to use official libgap APIs

### PR DESCRIPTION
**Pull request by fingolfin**
Addresses parts of #35361

### CODE REVIEW

**Comment by fchapoton** [comment]
This fails on garbage collection.

File: src/sage/libs/gap/libgap.pyx [hunk 1]
```diff
@@ -782,7 +782,7 @@ class Gap(Parent):
             sage: libgap.collect()
         """
         initialize()
-        rc = CollectBags(0, 1)
+        rc = GAP_CollectBags(1)
```
Comment (fchapoton) [inline src/sage/libs/gap/libgap.pyx]: maybe there is a difference in the returned code between these two functions ?

the existing one is declared as `UInt`, the new one as `void`

Comment (fchapoton) [reply src/sage/libs/gap/libgap.pyx]: should we just remove the check for the output code  below ?

Comment (fchapoton) [reply src/sage/libs/gap/libgap.pyx]: or we could wrap with `try ... except GAPError` maybe ?

Comment (fingolfin) [reply src/sage/libs/gap/libgap.pyx]: There is no return value, and the old function also never return anything other than success. It was a mirage. I'll remove it.

**Review by fchapoton — APPROVED** [review]
ok, thanks

**Comment by fingolfin** [comment]
There are all kinds of Conda errors now, but I think / hope they are not caused by this PR?

**Comment by fchapoton** [comment]
I have no idea. They seem rather to be of the standard "dependencies broken" style in the conda building.

**Pull request merged**